src/server/cdb/main.c

```diff
@@ -80,17 +80,24 @@ teardown(void)
     log_teardown();
 }
 
-static struct CDBHandle*
+static struct cdb_handle *
 setup_cdb_handle(cdb_options_st *opt)
 {
     cdb_setup();
 
-    char *cdb_file_path = opt->cdb_file_path.val.vstr;
-    if (cdb_file_path == NULL) {
+    struct cdb_handle_create_config cfg;
+    bstring_init(cfg.path);
+
+    cfg.load_method = (opt->use_mmap.val.vbool) ? CDB_MMAP : CDB_HEAP;
+
+    bstring_set_text(cfg.path, opt->cdb_file_path.val.vstr);
```

Comment:
have you tested this? i think `bstring_set_text` only works for string literals

Reply:
i think `bstring_set_raw` is for string literals. I've started up the server with this code and it answers queries.

Reply:
whoops, you are right. i misread.

Reply:
i had to try both :)

Reply:
some suggestions:

`bstring_set_raw` might be better named `bstring_set_literal`
`bstring_set_text` might be better named `bstring_set_char_p` or `bstring_from_char_ptr`

Reply:
I like `bstring_set_literal`. I think `bstring_set_cstr` is a clear name to me, not sure how you guys feel about it though.

Reply:
`bstring_set_cstr` is good. +1

Rust has `CStr`/`CString` to refer to a null-terminated string, so it follows that naming convention well.

Reply:
Created https://github.com/twitter/ccommon/issues/175 to update APIs as suggested